src/firebolt/client/client.py

```diff
@@ -54,7 +56,15 @@ class Client(FireboltClientMixin, HttpxClient):
 
     @cached_property
     def account_id(self) -> str:
-        return self.get(url=ACCOUNT_URL).json()["account"]["id"]
+        if self.account_name is not None:
+            response = self.get(
+                url=ACCOUNT_BY_NAME_URL, params={"account_name": self.account_name}
+            )
+            if response.status_code != 200:
```

Comment:
Let's raise AccountError only on 404 code.
In all other cases let's propagate the error further.
```
if response.status_code == httpx.codes.NOT_FOUND:
    raise AccountError(self.account_name)
# process all other status codes
response.raise_for_status()
```

Reply:
See query re Cursors with default account_id values above. Otherwise done.